soknad-felles/src/test/java/no/nav/k9/søknad/PeriodeTest.java

```diff
@@ -100,4 +104,70 @@ public void TestForsikreLukketPeriode() {
         assertEquals(LocalDate.parse("2020-02-01"), lukketPeriode.fraOgMed);
         assertEquals(LocalDate.parse("2020-02-03"), lukketPeriode.tilOgMed);
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void TestLeggTilPerioderMedDuplikater() {
+        var periode1 = Periode.parse("2020-01-01/2021-01-01");
+        var periode2 = Periode.parse("2020-04-04/..");
+        var periode3 = Periode.parse("2020-05-05/2020-06-06");
+
+        var eksisterende = Map.of(
+                periode1, true,
+                periode2, true
+        );
+        var nytt = Map.of(
+                periode1, true,
+                periode2, true,
+                periode3, true
+        );
+        leggTilPerioder(eksisterende, nytt);
+    }
+
+    @Test
+    public void TestLeggTilPerioderUtenDuplikater() {
+        var periode1 = Periode.parse("2020-01-01/2021-01-01");
+        var periode2 = Periode.parse("2020-04-04/..");
+        var periode3 = Periode.parse("2020-05-05/2020-06-06");
+
+        var eksisterende = new HashMap<>(Map.of(
+                periode1, true,
+                periode2, true
+        ));
+        var nytt = Map.of(
+                periode3, true
+        );
+
+        var oppdatert = leggTilPerioder(eksisterende, nytt);
```

Comment:
...siden "eksisterende" blir mutert bør den heller hete "perioder". I tillegg bør "eksisterende" tas vare på for senere sjekk. F.eks. gjennom:

```
var perioder = new HashMap<>(eksisterende);
var oppdatert = leggTilPerioder(perioder, nytt);
```
...tilsvarende gjelder på testene nedover også.